core/src/main/java/com/graphhopper/routing/GenericDijkstraBidirection.java

```diff
@@ -0,0 +1,212 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing;
+
+import com.carrotsearch.hppc.IntObjectMap;
+import com.graphhopper.coll.GHIntObjectHashMap;
+import com.graphhopper.routing.util.TraversalMode;
+import com.graphhopper.routing.weighting.Weighting;
+import com.graphhopper.storage.Graph;
+import com.graphhopper.storage.SPTEntry;
+import com.graphhopper.util.*;
+
+import java.util.PriorityQueue;
+
+/**
+ * Generic implementation of bidirectional Dijkstra algorithm that can be used with different shortest path entry types.
+ *
+ * @author Peter Karich
+ */
+public abstract class GenericDijkstraBidirection<T extends SPTEntry> extends AbstractBidirAlgo {
```

Comment:
Nice, probably we could merge with AbstractBidirAlgo and use GenericDijkstraBidirection for bidirectional A* too then.

Reply:
Yes I thought about this too. There should be quite a bit of code in Astar that can be reused here, but I have not looked close yet.